hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/NetworkTransactionGetReceiptHandler.java

```diff
@@ -84,23 +84,40 @@ public Response findResponse(@NonNull final QueryContext context, @NonNull final
         final var responseType = op.headerOrElse(QueryHeader.DEFAULT).responseType();
         if (header.nodeTransactionPrecheckCode() == OK && responseType != COST_ANSWER) {
-            final var history = recordCache.getHistory(transactionId);
+            final var topLevelTxnId = transactionId.nonce() > 0
+                    ? transactionId.copyBuilder().nonce(0).build()
+                    : transactionId;
+            final var history = recordCache.getHistory(topLevelTxnId);
             if (history == null) {
-                // Unlike with records, we only return RECEIPT_NOT_FOUND if we have never heard of this transaction.
+                // We only return RECEIPT_NOT_FOUND if we have never heard of this transaction.
                 responseBuilder.header(header.copyBuilder()
                         .nodeTransactionPrecheckCode(RECEIPT_NOT_FOUND)
                         .build());
             } else {
-                responseBuilder.receipt(history.userTransactionReceipt());
-                if (op.includeDuplicates()) {
-                    responseBuilder.duplicateTransactionReceipts(history.duplicateRecords().stream()
-                            .map(TransactionRecord::receiptOrThrow)
-                            .toList());
-                }
-                if (op.includeChildReceipts()) {
-                    responseBuilder.childTransactionReceipts(history.childRecords().stream()
-                            .map(TransactionRecord::receiptOrThrow)
-                            .toList());
+                // Only top-level transactions can have children and duplicates
+                if (transactionId == topLevelTxnId) {
+                    responseBuilder.receipt(history.userTransactionReceipt());
+                    if (op.includeDuplicates()) {
+                        responseBuilder.duplicateTransactionReceipts(history.duplicateRecords().stream()
+                                .map(TransactionRecord::receiptOrThrow)
+                                .toList());
+                    }
+                    if (op.includeChildReceipts()) {
+                        responseBuilder.childTransactionReceipts(history.childRecords().stream()
+                                .map(TransactionRecord::receiptOrThrow)
+                                .toList());
+                    }
+                } else {
+                    final var maybeRecord = history.childRecords().stream()
+                            .filter(record -> transactionId.equals(record.transactionID()))
+                            .findFirst();
+                    if (maybeRecord.isEmpty()) {
+                        responseBuilder.header(header.copyBuilder()
+                                .nodeTransactionPrecheckCode(RECEIPT_NOT_FOUND)
+                                .build());
+                    } else {
+                        responseBuilder.receipt(maybeRecord.get().receipt());
+                    }
```

Comment:
Seems worth extracting to a common method, but definitely not a PR blocker

Reply:
True Matt, noticed the duplicated pattern as well...then thought these queries aren't likely to change any time soon, and decided to leave it for now. 🤷